quant docs: add and clean up LayerNorm

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40377 quant docs: add and clean up ELU
* #40346 quant docs: add and clean up BatchNorm{n}d
* #40345 quant docs: add and clean up InstanceNorm{n}d
* #40343 quant docs: add and clean up GroupNorm
* **#40342 quant docs: add and clean up LayerNorm**
* #40341 quant docs: add and clean up hardtanh
* #40340 quant docs: add and clean up hardsigmoid
* #40323 quant docs: clean up hardswish

Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

Differential Revision: [D22152639](https://our.internmc.facebook.com/intern/diff/D22152639)

vkuzo committed:
quant docs: add and clean up LayerNorm

Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: add and clean up LayerNorm"


Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: add and clean up LayerNorm"


Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

Differential Revision: [D22152639](https://our.internmc.facebook.com/intern/diff/D22152639)

[ghstack-poisoned]

raghuramank100 commented on torch/nn/qat/modules/normalization.py:
```diff
@@ -182,10 +182,10 @@ def from_float(cls, mod, qconfig=None):
+    activation, used for quantization aware training.
 
-    Similar to `torch.nn.LayerNorm`, with FakeQuantize modules initialized to
+    Similar to :class:`torch.nn.LayerNorm`, with FakeQuantize modules initialized to
```
I think we do not need this class. By default, we insert an activation post process after the operation, without needing a new class.

raghuramank100 reviewed (changes requested):
Lets remove the QAT layer norm class.

vkuzo committed:
Update on "quant docs: add and clean up LayerNorm"


Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

Differential Revision: [D22152639](https://our.internmc.facebook.com/intern/diff/D22152639)

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: add and clean up LayerNorm"


Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

Differential Revision: [D22152639](https://our.internmc.facebook.com/intern/diff/D22152639)

[ghstack-poisoned]

vkuzo committed:
Update on "quant docs: add and clean up LayerNorm"


Cleans up the docstrings for quantized LayerNorm, and adds it to the docs.

Test plan:
* build on Mac OS and inspect

Differential Revision: [D22152639](https://our.internmc.facebook.com/intern/diff/D22152639)

[ghstack-poisoned]
